overlord/snapstate/snapstate.go

```diff
@@ -450,11 +464,8 @@ func revertToRevision(s *state.State, name string, rev snap.Revision) (*state.Ta
 	if i < 0 {
 		return nil, fmt.Errorf("cannot find revision %s for snap %q", rev, name)
 	}
-	revertToRev := snapst.Sequence[i].Revision
-
 	ss := &SnapSetup{
-		Name:     name,
-		Revision: revertToRev,
+		SideInfo: snapst.Sequence[i],
```

Comment:
this is a nice simplification, but now we need code to migrate this case :/ sorry I didn't think of that as I proposed it,

we need to go from name, revision in snapst.Sequence to the full thing, probably worth/correct only for tasks attached to non ready changes